resolwe_bio/tools/expression_aggregator.py

```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/env python2
+# pylint: disable=invalid-name,import-error
+"""Expression aggregator."""
+
+import argparse
+import json
+import csv
+import gzip
+
+import numpy as np
+
+
+def get_args():
+    """Parse command-line arguments."""
+    parser = argparse.ArgumentParser(description='Expression aggregator')
+    parser.add_argument('-e', '--expressions', nargs='+', help='Expressions', required=True)
+    parser.add_argument('-d', '--descriptors', nargs='+', help='Descriptors', required=True)
+    parser.add_argument('-t', '--expression-types', nargs='+', help='Expression types')
+    parser.add_argument('-a', '--aggregator', help='Aggregator', required=False)
+    parser.add_argument('-b', '--output_box_plot', help='Output box plot', required=False)
+    parser.add_argument('-x', '--output_expressions', help='Output expressions', required=False)
+    return parser.parse_args()
+
+
+def get_expression(fname=None, sep='\t'):
+    """Read expressions from file."""
+    with gzip.open(fname, 'rt') as f:
+        reader = csv.DictReader(f, delimiter=sep)
+        return {row['Gene']: float(row['Expression']) for row in reader}
+
+
+def get_indices(my_list, my_element):
+    """Return positions of my_element in my_list."""
+    return {i for i, x in enumerate(my_list) if x == my_element}
+
+
+def get_values(expressions, descriptors, gene, descriptor):
+    """Return expressions of a gene with matching descriptor."""
+    indices = get_indices(descriptors, descriptor)
+    return [expression[gene] for i, expression in enumerate(expressions) if i in indices and gene in expression]
+
+
+def get_expressions(fnames=[], sep='\t', descriptors=[]):
+    """Read expressions from files."""
+    expressions = [get_expression(fname, sep) for fname in fnames]
+    genes = set().union(*[expression.keys() for expression in expressions])
+    return {gene: {descriptor: get_values(expressions, descriptors, gene, descriptor)
```

Comment:
return {
        gene: {
            descriptor: get_values(expressions, descriptors, gene, descriptor)
            for descriptor in set(descriptors)
            if get_values(expressions, descriptors, gene, descriptor)
        } 
        for gene in genes
    }